src/App.tsx

```diff
@@ -1,61 +1,126 @@
 import './App.scss';
 
-// import usersFromServer from './api/users';
-// import todosFromServer from './api/todos';
+import usersFromServer from './api/users';
+import todosFromServer from './api/todos';
+
+import React, { useState } from 'react';
+
+import { TodoList } from './components/TodoList';
+import { User } from './components/UserInfo';
+
+export type Todo = {
+  id: number;
+  title: string;
+  userId: number;
+  completed: boolean;
+  user: User | null;
+};
+
+function getUserById(userId: number) {
+  return usersFromServer.find(user => user.id === userId) || null;
+}
+
+export const todos = todosFromServer.map(todo => ({
+  ...todo,
+  user: getUserById(todo.userId),
+}));
+
+function getTodoId(initialTodos: Todo[]) {
+  const todoArrId = initialTodos.map(todo => todo.id);
+
+  return Math.max(...todoArrId) + 1;
+}
+
+export const App: React.FC = () => {
+  const [titleValue, setTitleValue] = useState('');
+  const [titleError, setTitleError] = useState(false);
+  const [userValue, setUserValue] = useState(0);
+  const [userError, setUserError] = useState(false);
+  const [currentTodos, setCurrentTodo] = useState(todos);
+
+  const addTodo = (newTodo: Todo) => {
+    setCurrentTodo((prevTodos: Todo[]) => [...prevTodos, newTodo]);
+  };
```

Comment:
[CODE STYLE] - The `addTodo` function is currently directly manipulating the state by taking the previous state and appending a new todo. This is fine, but make sure to always treat the state as immutable to prevent any unexpected behavior. Your current implementation does correctly create a new array which is good.